platform/linux-generic/pktio/netmap.c

```diff
@@ -388,13 +389,22 @@ static int netmap_open(odp_pktio_t id ODP_UNUSED, pktio_entry_t *pktio_entry,
 
 	if (pkt_nm->is_virtual) {
 		static unsigned mac;
+		uint32_t tid = syscall(SYS_gettid);
+
+		if ((int)tid == -1)
```

Comment:
man gettid says that it always passes.

kernel code is also:
pid_t __task_pid_nr_ns(struct task_struct *task, enum pid_type type,
			struct pid_namespace *ns)
{
	pid_t nr = 0;

	rcu_read_lock();
	if (!ns)
		ns = task_active_pid_ns(current);
	if (likely(pid_alive(task))) {
		if (type != PIDTYPE_PID)
			task = task->group_leader;
		nr = pid_nr_ns(rcu_dereference(task->pids[type].pid), ns);
	}
	rcu_read_unlock();

	return nr;
}
EXPORT_SYMBOL(__task_pid_nr_ns);

I.e. it will return init process tid 0 in worst case. So this check is not correct and not needed.

It might be needed additional cast:
uint32_t tid = (uint32_t)syscall(SYS_gettid) because of syscall returns pid_t.

Reply:
.

Reply:
syscall() specification still defines -1 as an error return value and we should adhere to the spec regardless of the function implementation which may change.

Reply:
@matiaselo yes, so this is fine.

Reply:
@matiaselo @lumag  yep, I forgot that some syscalls might be not implemented on some systems, that  also will return error.